superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx

```diff
@@ -284,7 +284,7 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
         extra={<Error>{filterState.validateMessage}</Error>}
       >
         <StyledSelect
-          allowClear={!enableEmptyFilter}
+          allowClear
```

Comment:
As we're already validating that the filter is set before letting the user apply the filter selection, we no longer need to disable `allowClear`.